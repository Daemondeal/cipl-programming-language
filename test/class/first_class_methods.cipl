class Test:
    set_greeting(greeting):
        this.greeting = greeting

    println_greeting():
        println(this.greeting)

let instance = Test()
instance.set_greeting("Hello World!")
instance.println_greeting()

let instance2 = Test()
instance2.set_greeting("Hi there!")
instance2.println_greeting()

instance2.stolen_greeting = instance.println_greeting
instance2.stolen_greeting() # Should println "Hello World"