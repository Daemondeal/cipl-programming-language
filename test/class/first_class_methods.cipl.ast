(class Test
    (proc set_greeting (params greeting) (body
        (expression (set greeting greeting this))
    ))
    (proc println_greeting (body
        (expression (call println (get greeting this)))
    ))

)
(let instance (call Test))
(expression (call (get set_greeting instance) Hello World!))
(expression (call (get println_greeting instance)))
(let instance2 (call Test))
(expression (call (get set_greeting instance2) Hi there!))
(expression (call (get println_greeting instance2)))
(expression (set stolen_greeting (get println_greeting instance) instance2))
(expression (call (get stolen_greeting instance2)))

