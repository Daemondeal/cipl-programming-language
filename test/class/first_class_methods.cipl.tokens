Token          Lexeme         Line 
CLASS          class          1    
IDENTIFIER     Test           1    
COLON          :              1    
LINE_END       \n             2    
INDENT                        2    
IDENTIFIER     set_greeting   2    
LEFT_PAREN     (              2    
IDENTIFIER     greeting       2    
RIGHT_PAREN    )              2    
COLON          :              2    
LINE_END       \n             3    
INDENT                        3    
THIS           this           3    
DOT            .              3    
IDENTIFIER     greeting       3    
EQUAL          =              3    
IDENTIFIER     greeting       3    
LINE_END       \n             4    
DEDENT                        5    
IDENTIFIER     println_greeting5    
LEFT_PAREN     (              5    
RIGHT_PAREN    )              5    
COLON          :              5    
LINE_END       \n             6    
INDENT                        6    
IDENTIFIER     println        6    
LEFT_PAREN     (              6    
THIS           this           6    
DOT            .              6    
IDENTIFIER     greeting       6    
RIGHT_PAREN    )              6    
LINE_END       \n             7    
DEDENT                        8    
DEDENT                        8    
LET            let            8    
IDENTIFIER     instance       8    
EQUAL          =              8    
IDENTIFIER     Test           8    
LEFT_PAREN     (              8    
RIGHT_PAREN    )              8    
LINE_END       \n             9    
IDENTIFIER     instance       9    
DOT            .              9    
IDENTIFIER     set_greeting   9    
LEFT_PAREN     (              9    
STRING         "Hello World!" 9     Hello World!
RIGHT_PAREN    )              9    
LINE_END       \n             10   
IDENTIFIER     instance       10   
DOT            .              10   
IDENTIFIER     println_greeting10   
LEFT_PAREN     (              10   
RIGHT_PAREN    )              10   
LINE_END       \n             11   
LET            let            12   
IDENTIFIER     instance2      12   
EQUAL          =              12   
IDENTIFIER     Test           12   
LEFT_PAREN     (              12   
RIGHT_PAREN    )              12   
LINE_END       \n             13   
IDENTIFIER     instance2      13   
DOT            .              13   
IDENTIFIER     set_greeting   13   
LEFT_PAREN     (              13   
STRING         "Hi there!"    13    Hi there!
RIGHT_PAREN    )              13   
LINE_END       \n             14   
IDENTIFIER     instance2      14   
DOT            .              14   
IDENTIFIER     println_greeting14   
LEFT_PAREN     (              14   
RIGHT_PAREN    )              14   
LINE_END       \n             15   
IDENTIFIER     instance2      16   
DOT            .              16   
IDENTIFIER     stolen_greeting16   
EQUAL          =              16   
IDENTIFIER     instance       16   
DOT            .              16   
IDENTIFIER     println_greeting16   
LINE_END       \n             17   
IDENTIFIER     instance2      17   
DOT            .              17   
IDENTIFIER     stolen_greeting17   
LEFT_PAREN     (              17   
RIGHT_PAREN    )              17   
LINE_END       \n             17   
EOF                           17   
