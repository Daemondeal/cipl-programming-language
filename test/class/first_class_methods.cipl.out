::stdout::
Hello World!
Hi there!
Hello World!

::stderr::
